README.md

```diff
@@ -153,19 +155,158 @@ Headers may be specified globally for all files in the bucket by adding a `heade
+          replaceKeyWith: [object]
+        condition:
+          keyPrefixEquals: [prefix]
+          httpErrorCodeReturnedEquals: [CODE]
+      - [more-rules...]
+    ...
+```
+
+The `routingRules` option can be used to define rules for when and how certain requests to your site should be redirected. Each rule in the `redirectRules` list consists of a (required) `redirect` definition and (optionally) a `condition` on which the redirect is applied.
+
+The `redirect` property of each rule has five optional parameters:
+
+* `hostName` is the name of the host that the request should be redirected to (e.g. 'www.other-site.com'). Defaults to the host from the original request.
+* `httpRedirectCode` is the HTTP status code to use for the redirect (e.g. 301, 303, 308).
+* `protocol` is the protocol to use for the redirect and must be 'http' or 'https'. Defaults to the protocol from the original request.
+* `replaceKeyPrefixWith` specifies the string to replace the portion of the route specified in the `keyPrefixEquals` with in the redirect. For instance, if you want to redirect requests for pages starting with '/images' to pages starting with '/assets/images', you can specify `keyPrefixEquals` as '/images' and `replaceKeyPrefixWith` as '/assets/images'. _Cannot be specified along with `replaceKeyWith`_.
+* `replaceKeyWith` specifies a specific page to redirect requests to (e.g. 'landing.html'). _Cannot be specified along with `replaceKeyPrefixWith`_.
+
+The `condition` property has two optional parameters:
+
+* `keyPrefixEquals` specifies that requests to pages starting with the specified value should be redirected. Often used with the `replaceKeyPrefixWith` and `replaceKeyWith` `redirect` properties.
+* `httpErrorCodeReturnedEquals` specifies that requests resulting in the given HTTP error code (e.g. 404, 500) should be redirected.
+
+_If `condition` is not specified, then all requests will be redirected in accordance with the specified `redirect` properties_
+
+[AWS Documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-websiteconfiguration.html#cfn-s3-websiteconfiguration-routingrules)
 
 ---
 
 ### Command-line Parameters
 
-No command-line parameters exist at this time.
+**--region**
+
+_optional_, defaults to value specified in `provider` section of `serverless.yml`
+
+```bash
+serverless client deploy --region $REGION
+```
+
+Use this parameter to specify what AWS region your bucket will be deployed in.
+
+---
+
+**--no-delete-contents**
+
+_optional_, default `false` (deletes contents by default)
+
+```bash
+serverless client deploy --no-delete-contents
+```
+
+Use this parameter if you do not want to delete the contents of your bucket before deployment. Files uploaded during deployment will still replace any corresponding files already in your bucket.
+
+**Note:** *If this option is specified, your bucket will also not be deleted before deployment (by default the bucket is removed before deployment).*
```

Comment:
@linusmarco Did this change? I thought we only deleted the bucket when running `sls client remove`?

Did you mean that we wont delete the contents here?

Btw - documentation looks wayyyy better than it used to. I'm still slugging through the code changes and will do my best to get you a review soon.

Reply:
Very good catch! somehow I thought that we did delete the bucket before a deploy, but we didn't. The new code DOES, though, which I should change back. I'll make that update in the code and docs.

Reply:
@linusmarco I'm not necessarily opposed to it - Is there a particular reason for it though? 

My only concern might be that I've seen/heard of weird caching/delays on S3 bucket deletion so I would worry we might run into some case when we delete it and then can't recreate it for some reason.

Reply:
@fernando-mc There wasn't any real intentionality behind this on my part--I for some reason I just mis-translated this bit of functionality. I share your concerns about delays, so I think it's probably best to switch back to the old way.

Reply:
Sounds good! Thanks!